### DISCUSSION TITLE
Linux Support

### DISCUSSION THREAD

**Discussion by buraktabn** [JaffaKetchup/flutter_map_tile_caching#75] [Q&A] [answered]
Is there a plan for adding support for Linux?

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:294:7)
<asynchronous suspension>
#1      getApplicationDocumentsDirectory (package:path_provider/path_provider.dart:115:24)
<asynchronous suspension>
#2      RootDirectory.normalCache (package:flutter_map_tile_caching/src/root/directory.dart:50:8)
<asynchronous suspension>
#3      main (package:renp/main.dart:17:36)
<asynchronous suspension>
```

**Comment by buraktabn** [comment]
Well, noticed that it couldn't find a directory to save it with the following code.

```dart
FlutterMapTileCaching.initialise(await RootDirectory.temporaryCache);
```

It works with custom `RootDirectory`.

```dart
final tempDir = await getTemporaryDirectory();

FlutterMapTileCaching.initialise(RootDirectory.custom(tempDir));
```

**Comment by JaffaKetchup** [comment, marked as answer]
Hey @buraktabn,

`RootDirectory.normalCache` and `RootDirectory.temporaryCache`use `getApplicationDocumentsDirectory` and `getTemporaryDirectory` (from '[path_provider](https://pub.dev/packages/path_provider)') respectively behind the scenes. As you can see from the official table below, both of these should be supported by 'path_provider' - indeed, if you are using `getTemporaryDirectory` like in your [second example](https://github.com/JaffaKetchup/flutter_map_tile_caching/discussions/75#discussioncomment-4124999), that's 'path_provider'.
![image](https://user-images.githubusercontent.com/58115698/201493131-1faebb4d-4b8a-49d2-85f8-47d0aa2b593a.png)

The error you had may occur when installing this package before a full app restart. A full app restart usually resolves issues similar to that one.

Note you may run into two issues with your implementation:
1. Using `temporaryCache`/`getTemporaryDirectory` usually isn't recommended because they can be cleared by the OS at any time. This would lose all stored information and tiles.
2. You haven't used '/<subdir>' on the end of your path, meaning that this project will pollute the temp directory more than usual.

If this library has helped you in a commercial situation, please consider [donating](https://github.com/sponsors/JaffaKetchup) - every little helps! If you can't please leave a star and a like :)
